src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/AutoCrafterListener.java

```diff
@@ -58,6 +65,21 @@ public void onInteract(PlayerRightClickEvent e) {
                     return;
                 }
 
+                // Check if the recipe of the item is disabled.
+                boolean unlocked = true;
+                for (Recipe recipe : Bukkit.getRecipesFor(e.getItem())) {
+                    if (!e.getPlayer().hasDiscoveredRecipe(((Keyed) recipe).getKey())) {
```

Comment:
Unsafe class cast, you never check if the recipe is actually an instance of `Keyed` here.

Reply:
Okay. Should I add an && checking it then?